### ISSUE TITLE
Angular 15 support

### ISSUE THREAD

**Issue by jerone** [AzureAD/microsoft-authentication-library-for-js#5410]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

Latest

### Wrapper Library

MSAL Angular (@azure/msal-angular)

### Wrapper Library Version

Latest

### Public or Confidential Client?

Public

### Description

Will MSAL support Angular 15.

### Source

External (Customer)

**Comment by YanerTavuz** [comment]
i think that since Ivy is required in Angular v15, then this is connected to https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/4298 maybe?

**Comment by sameerag** [comment]
@jerone We are in the process of adding Angular 15 support. We have started tracking it internally and someone will start looking into this actively soon. I do not have a specific ETA yet but will be prioritized as soon as we can. cc @EmLauber

**Comment by peterzenz** [comment]
Assigning to @EmLauber for tracking.  This should get picked up in Q1 2023.

**Comment by aylmercarson** [comment]
stopped me dead in my tracks after promoting azure B2C and Angular. hard to believe that MS can't/won't keep up with the NG release cycle given the HUGE number of corporate integrations there are.

**Comment by jerone** [comment]
Still relevant!

**Comment by kauppfbi** [comment]
@tnorling @peterzenz @EmLauber, 

could you please provide some transparency and share your thoughts and plans on this topic? 
I assume for many enterprises, this is a major blocker right now. In our case, the `@azure/msal-angular` package is the last one, which is shipped with `ViewEngine`. It not only blocks us from updating to v15, it also prevents us from using `pnpm` with recommended settings. 

It seems like effort and help from the community is not really requested/appreciated. So I would at least be interested in your timeline or maybe there are some dependencies to other topics, which the community is not aware of. 
Should I be wrong and you would appreciate the help of the community on this topic, just tell us. I guess there are many devs - including me - who are happy to help.

Thank you very much for your efforts.

**Comment by tnorling** [comment]
@kauppfbi I apologize if anyone gave the impression that we do not appreciate external contributions, that could not be further from the truth. We would love to see more community engagement and would welcome PRs that will help unblock your scenarios. To set expectations, however, just because someone opens a PR doesn't mean that it will be accepted. There are many things that we need to consider and keep in mind, especially when it may involve breaking changes. We commit to reviewing all PRs and providing feedback and if the author is able to address that feedback we will be delighted to merge it.

As for a timeline, Peter did share one [above](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/5410#issuecomment-1361757895)

**Comment by gaweCodes** [comment]
Our projects are also blocked. It would be great to have a solution asap. Can someone please provide an update on the release date for this issue? Thank you.

**Comment by xXanth0s** [comment]
Same here. 
Our project is blocked because of this dependency. 
Are there any information about the Angular 15 support?

**Comment by EmLauber** [comment]
The timeline is the same as above. We have this on our roadmap to get picked up in Q1 2023.

**Comment by aylmercarson** [comment]
from the docs: "At a minimum, @azure/msal-angular will follow the [support schedule of the main Angular project](https://angular.io/guide/releases#support-policy-and-schedule)."

angular 15 was released 2 months ago at the date of writing this. i am beginning to wonder if the birth of blazor as a competing technology is affecting the willingness of ms to keep to its word and keep up with the angular release cycle?

**Comment by jerone** [comment]
Also, be aware that v16.0 is already [planned for week of 2023-05-01](https://angular.io/guide/releases#release-schedule). 
Please, please, please put support for v16.0 on your roadmap, so you and us can prevent this kind of frustrations.
🙏🙏🙏

**Comment by gaweCodes** [comment]
It's really tedious. Many Angular apps have dependencies on small npm packages. The community or the developers behind these small packages often have more will to keep up with the Angular release cycle than the big players like here. I don't know why this has to take so long.
By the way: A release in Q1 2023 is not really a date. I would like to know if it is still coming in January or at the end of March?

**Comment by DanielRose** [comment]
> The timeline is the same as above. We have this on our roadmap to get picked up in Q1 2023.

That is not even a promise of a release in Q1 2023, just that someone will start looking at it in Q1 2023. In other words, there is no assurance at all of when this might be released.

Angular 9, which contained the rollout of Ivy, was released on February 2020. The opt-in preview was already in the beta of June 2019. The official deprecation was in Angular 12, released May 2021 (which included the message "Now that View Engine is deprecated, it will be removed in a future major release").

This is something which was a very long time coming, and I don't understand how this was apparently forgotten or at least not given any priority.

**Comment by EmLauber** [comment]
Thank you to everyone for your feedback on this. I recognize MSAL's support of Angular 15 delays your own efforts of updating to Angular 15. We do take this feedback seriously and are discussing internally how better to prioritize this customer need in the future. 

To some of the specific comments:
> coming in January

I have no updates on a specific release date, but it won't be coming in January. 

>  blazor as a competing technology

Blazor's development is exciting, but not related to MSAL.browser's prioritization. These two projects release on independent schedules.

>  put support for v16.0 on your roadmap

Thank you for the feedback. We are tracking the v16 Angular release schedule. I don't have a specific release date for MSAL.browser's support for v16, but we are aware of it.

**Comment by sandrooco** [comment]
@EmLauber I (and probably a lot of other people) appreciate the bits of communication you provided in the last weeks.
Respectfully, I'm wondering what causes this massive delay? Can you elaborate for us?

You might ask why everyone is unpleased (...) with the current situation.
A few inputs from the community's perspective: the technology the community wants support for has been around for quite some time, the Angular team provided lots of info about migrating and their timeline. Some people provided PRs that provided (at least) baseline support. These weren't merged because it breaks support for Angular version that not even Angular itself supports anymore. If it's possible for me to provide a PR in 4h (including setup for MS' OSS requirements) it seems like one of you could do this easily in 1h. Obviously it's not always that easy - maybe there are actual reasons that the community simply never has heard of.

Thank you for keeping us in the loop.

**Comment by EmLauber** [comment]
We appreciate the community PRs, but as you called out, they broke previous Angular version support within the MSAL.js library. We have an engineer this quarter actively looking at how best to move forward with adding Angular 15 support while addressing the impact to MSAL customers who may not have upgraded to the latest Angular version yet.

**Comment by ghost** [comment]
This is why you keep separate versions of your package that explicitly support a specific Angular version so that you can safely make breaking changes while keeping a usable version for older Angular versions.

**Comment by DM-be** [comment]
any update on an ETA? We are also waiting to update to angular 15 on this...

**Comment by EmLauber** [comment]
We have an engineer working on this issue this quarter and will update here once we have updated ETA.

**Comment by ZeroGDrive** [comment]
https://github.com/angular/angular-cli/pull/24720

**Comment by Lonli-Lokli** [comment]
@EmLauber Could you forward this link to the engineer? https://github.com/auth0/auth0-angular/issues/255#issuecomment-1261795471

**Comment by jerone** [comment]
Time to update the changelog; the ["mid-term"](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/roadmap.md#mid-term) is already here.

**Comment by kauppfbi** [comment]
Angular 16 is in active development. Just mentioning at this stage. 

https://github.com/angular/angular/releases/tag/16.0.0-next.0

**Comment by ThoSap** [comment]
@EmLauber sorry to bug you, do you have an update for us?

**Comment by EmLauber** [comment]
Our engineer is still working on this. We don't yet have a finalized release date.

**Comment by ikeppu** [comment]
> Our engineer is still working on this. We don't yet have a finalized release date.

Any information about release date ?

**Comment by YanerTavuz** [comment]
I don't quite understand why people are so stressed about this release. 
The package owners have explained why it is taking time and still people are not understanding. 

Here is my take on the issue: 

Currently the package is built for View Engine which supports multiple versions of Angular, however Angular v15+ has dropped this support. 
So why can't the package owners just change targeting to Angular versions that supports Ivy? 
Well, because not everybody upgrades their Angular versions, and if the "flow" of the login process changes then older versions of Angular (that only uses View Engine) still needs to be able to use newer versions of this package so they can also use the new flows. 

The problem that the package owners/developers are facing is how to maintain two different versions. One that supports View Engine, and one that supports Ivy. 

Correct me if I'm wrong 🤓

**Comment by sandrooco** [comment]
@YanerTavuz "The package owners have explained why it is taking time and still people are not understanding." It's correct, I don't understand. It hasn't been explained though.

Maintaining multiple versions is easy, see Vue and/or Spring for example. They still have security updates for the older 2.x versions but mainly work on 3.x.

**Comment by YanerTavuz** [comment]
Hmm, well. Now that you explain it like that 😅 @sandrooco  
So kind of like keeping old major version for legacy patches and newer major versions for Ivy supported only

**Comment by Lonli-Lokli** [comment]
I think the main reason is lack of development resources, so can propose 2 options
- hire me as a software engineer, I will fix it
- ask for chatgpt4 to create a fix, sign cla and release it

**Comment by DanielRose** [comment]
> Currently the package is built for View Engine which supports multiple versions of Angular, however Angular v15+ has dropped this support. So why can't the package owners just change targeting to Angular versions that supports Ivy? Well, because not everybody upgrades their Angular versions, and if the "flow" of the login process changes then older versions of Angular (that only uses View Engine) still needs to be able to use newer versions of this package so they can also use the new flows.
> 
> The problem that the package owners/developers are facing is how to maintain two different versions. One that supports View Engine, and one that supports Ivy.

Ivy exists since Angular 9, which was released in February 2020. From https://angular.io/guide/releases:

"All major releases are typically supported for 18 months. [...] Angular versions v2 to v12 are no longer under support."

So this shouldn't be a problem.

**Comment by schmkr** [comment]
Maybe a suggestion if maintenance and versioning is a concern: 

Recently Sentry was facing similar challenges. They had/have their `@sentry/angular` package and do want to keep on releasing that one (with ViewEngine support) with all new features they keep on adding to their product. So they published an entirely new package `@sentry/angular-ivy`. See https://github.com/getsentry/sentry-javascript/issues/5268#issuecomment-1446489347.

**Comment by EmLauber** [comment]
I appreciate everyone's engagement and suggestions. The team has been working on a new major MSAL release (v3) which will use the Ivy engine and support Angular 15. This is actively in development. I will update here once released and those interested in Ivy and Angular 15 can update to v3.

**Comment by jerone** [comment]
[**@jerone**](https://github.com/jerone) commented on [Jan 20, 2023, 10:36 AM GMT+1](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/5410#issuecomment-1398135667 "2023-01-20T09:36:37Z - Replied by Github Reply Comments"):
> Also, be aware that v16.0 is already [planned for week of 2023-05-01](https://angular.io/guide/releases#release-schedule). 
> Please, please, please put support for v16.0 on your roadmap, so you and us can prevent this kind of frustrations.
> 🙏🙏🙏

Less then 1 month away from completely skipping Angular 15 and going straight to version 16.

_FYI. this is literally the only package that is keeping our products from upgrading to the latest Angular version._

**Comment by aylmercarson** [comment]
something about a dead horse and flogging...i gave up on this and went with 

https://github.com/damienbod/angular-auth-oidc-client

it's fine and does all i want/need...

**Comment by rubiesonthesky** [comment]
It's possible also to use general browser package in Angular app. Of course depending on your needs but that is how we are using this library with Angular 15.

https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-browser

**Comment by stevenruizhang** [comment]
any update on an ETA? We are also waiting when v3 can support ng 15

**Comment by RichardCutmore** [comment]
Is there any schedule for release?

**Comment by nelsonmontano** [comment]
Is there any schedule for release?

**Comment by EmLauber** [comment]
We are targeting an alpha release of v3 this month (April).

**Comment by sandrooco** [comment]
For those interested, this is the pull request (merged): https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5753. Not sure about the difference to my (and other) PR.
At least this thing is about to land soon.

**Comment by Alexiafe** [comment]
> 

Any update yet ?

**Comment by tonycarew** [comment]
> We are targeting an alpha release of v3 this month (April).

Can you clarify which April?

**Comment by eluchsinger** [comment]
So to summarize, Angular 16 is scheduled for release this month and we are still stuck with Angular 14 because of this library due to a bug that was raised in 2021 (#4298).

It looks like the fix is not coming anytime soon, because we are still talking about an (obviously delayed) Alpha, while this library is being used in many production applications. To go from Alpha to a GA release, I again expect months. And because it's bound to a major release v3, I also expect breaking changes, major refactorings and major bugs.

I will start looking for an alternative now.

**Comment by kauppfbi** [comment]
Actually, Angular 16 will already be released this week. 

Edit: https://github.com/angular/angular/releases

**Comment by EmLauber** [comment]
> Can you clarify which April?

Yes the target was April for an alpha release for v3, but we had some unexpected delays push it back. The team is actively working on a release, and I will update here once it is available. 

> major release v3, I also expect breaking changes, major refactorings and major bugs.

We aimed to keep breaking changes to a minimum for v2 to v3. We have not yet released, but you can see the expected changes [here ](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/v2-v3-upgrade-guide.md)

> Angular 16 will already be released this week

We are actively evaluating v3 against Angular 16 and will provide an update here and on the README once we confirm it is compatible.

**Comment by EmLauber** [comment]
MSAL.Angular v3 alpha with Angular 15 and Ivy support is released: https://github.com/AzureAD/microsoft-authentication-library-for-js/releases/tag/msal-angular-v3.0.0-alpha.0

**Comment by grosch-intl** [comment]
@EmLauber I keep getting errors about package "" referring to a non-existing file.  What should I be putting in my package.json to use this version?

**Comment by YanerTavuz** [comment]
@grosch-intl 
If you look at the package in npmjs https://www.npmjs.com/package/@azure/msal-angular/v/3.0.0-alpha.0, you can see that you should add this to package.json
`"@azure/msal-angular": "3.0.0-alpha.0"`

**Comment by Kahera** [comment]
That's fantastic news! Is there any timeline for when we could expect it to be out of alpha?

**Comment by jornare** [comment]
Angular 16 is out now, will this new release also work fine with it?

**Comment by grosch-intl** [comment]
> If you look at the package in npmjs https://www.npmjs.com/package/@azure/msal-angular/v/3.0.0-alpha.0, you can see that you should add this to package.json `"@azure/msal-angular": "3.0.0-alpha.0"`

@YanerTavuz When doing just @azure/msal-angular that works.  But then I get a warning about msal-browser having an incorrect peer dependency as it wants 3.0.0-alpha.0 as well.  Trying to use that then gives the error I showed.

**Comment by YanerTavuz** [comment]
@grosch-intl 
Then add both packages with that version to package.json. I did and it worked.

```
"@azure/msal-angular": "3.0.0-alpha.0",
"@azure/msal-browser": "3.0.0-alpha.0"
```

**Comment by grosch-intl** [comment]
Switching from yarn to npm and looking at the package-lock.json I see this in the output:

```json
    "node_modules/@azure/msal-browser": {
      "version": "3.0.0-alpha.0",
      "resolved": "https://registry.npmjs.org/@azure/msal-browser/-/msal-browser-3.0.0-alpha.0.tgz",
      "integrity": "sha512-LHzFi1TI2xWMqmpV15Lneajge21ZDlX5t6eyfHXs9yLxNeccBwJTN7a2pFfXSX3ADeXy22KI9D0XN/zrAHcFLA==",
      "bundleDependencies": [
        "@azure/msal-common"
      ],
      "dependencies": {
        "@azure/msal-common": "file:../msal-common"
      },
      "engines": {
        "node": ">=0.8.0"
      }
    },
```

I don't think that `file:` thing is supposed to be there, is it?

**Comment by grosch-intl** [comment]
This update allows the `provideRouter(routes)` to work now but still doesn't work with `provideHttpClient`. Is that something you're still working on?

**Comment by Lonli-Lokli** [comment]
> I think the main reason is lack of development resources, so can propose 2 options
> 
> * hire me as a software engineer, I will fix it
> * ask for chatgpt4 to create a fix, sign cla and release it

Just a friendly reminder

**Comment by EmLauber** [comment]
Thanks for the quick feedback. We are aware that the alpha release is not working with some package managers like yarn and working on a solution. CC @konstantin-msft for when we have an update.

**Comment by EmLauber** [comment]
> Is there any timeline for when we could expect it to be out of alpha?

Timeline to move out of alpha will depend on the feedback we receive on the alpha version. People testing and reporting issues such as above helps this.

> Angular 16 is out now, will this new release also work fine with it?

We are still testing the alpha release against Angular 16.

**Comment by mrivero2020** [comment]
Can someone explain the benefits of using the @azure/msal-angular library over the @azure/msal-browser library directly in an Angual SPA? Is it that @azure/msal-angular just hides some complexity from using @azure/msal-browser directly, but it does not add other features? My understanding is that @azure/msal-angular is just a repackaging of @azure/msal-browser, and at least someone left a comment above that for vanilla authentication scenarios the @azure/msal-browser library works fine.

**Comment by Lonli-Lokli** [comment]
You can check source code itself https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-angular/src

**Comment by jerone** [comment]
This is **not** going to be a fun upgrade 😟

1. Angular 15 upgrade
3. Angular 16 upgrade
4. MSAL Browser major upgrade; meaning breaking changes to test.

**Comment by touhidrahman** [comment]
Did anyone have success with Angular 16 and msal 3.0.o-alpha.0? I would greatly appreciate knowing how you managed it, I am hanging in here.

**Comment by sithwarrior** [comment]
Im very confused by all of this..

When I checkout a project, I do it here on GitHub, and according to the project readme, the current version is msal-angular 3, which supports Angular 15.

[https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-angular](url)

However.. in reality there is only a alpha available??
And it's up in the air, if this version supports Angular 16.. the current version of Angular.

If I run the install command of the project

`npm install @azure/msal-browser @azure/msal-angular@latest`

I get 2.57.. not 3.x

Its May.. Angular 15 was released in November.. 6 months ago.
The general removal of NGCC was communicated a very long time ago

**Comment by touhidrahman** [comment]
For anyone who is looking for Angular 16 and msal 3 integration, [here is the sample project PR](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5944)

**Comment by sandrooco** [comment]
@sithwarrior @latest will always be the current stable build. Check the [table on npm](https://www.npmjs.com/package/@azure/msal-angular?activeTab=versions), it might clarify. To get the alpha, you need to install it like this: `npm install @azure/msal-browser@alpha @azure/msal-angular@alpha`

**Comment by sandrooco** [comment]
@EmLauber Regarding testing and providing feedback: is there a certain place where we should put the feedback? Is https://github.com/AzureAD/microsoft-authentication-library-for-js/discussions/5977 the right place?
So far all has been good for me with the alpha using npm and Angular 16.

**Comment by grosch-intl** [comment]
> For anyone who is looking for Angular 16 and msal 3 integration, [here is the sample project PR](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5944)

Would be great if the sample project used more of the newer angular features.  This is still all module based instead of standalone, for example.

**Comment by DM-be** [comment]
@grosch-intl 

you can bootstrap the app component as standalone and extend the msalredirectcomponent.

all you need is a super call with the msalauthservice injected in the app component after extending.

everything works as intended like that, using the <app-redirect> :)

**Comment by grosch-intl** [comment]
@DM-be I'm not sure what you mean by extend the msalredirectcomponent.  However, if the point of this is to be an Angular 16 example, shouldn't it include things that have existed since Angular 14?  standalone, using `inject` instead of passing things to a constructor, etc...

**Comment by Lonli-Lokli** [comment]
@grosch-intl I believe you can send your PR with these changes?

**Comment by DM-be** [comment]
@grosch-intl 

I mean extending the appcomponent:

export class AppComponent extends MsalRedirectComponent 
....

ctor:
super(inject(MsalService));

thats all there is to it.

Also, inject and standalone are still optional features of v16

**Comment by EmLauber** [comment]
> is there a certain place where we should put the feedback?

Any new bugs or feedback for the alpha release should be created as [new issues](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/new/choose) instead of commented on the release notes. Issues are easier to debug, catalogue, and resolve compared to comments. We can engage better when issues are focused on a particular topic by requiring details such as version, public/confidential, etc.

> Angular 16 sample 

Yes you can see the [PR](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5944) the team is working on for an Angular 16 sample. If you'd like, you can "Subscribe" directly to the PR and be notified immediately when it is merged instead of waiting for me to update this thread.

**Comment by mtosic** [comment]
> > Angular 16 sample
> 
> Yes you can see the [PR](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5944) the team is working on for an Angular 16 sample. If you'd like, you can "Subscribe" directly to the PR and be notified immediately when it is merged instead of waiting for me to update this thread.

Hi @EmLauber, I wanted to create an issue, but there is no option for selecting v3 of MSAL-Angular.

As for Angular 16 sample, it would be great if you could also make "standalone" sample. In Angular v16, there is even an option in CLI "ng new --standalone" which completely removes modules when creating new Angular app.

**Comment by rafagsiqueira** [comment]
@mtosic I did it with standalone this way. Sharing it here, since it might help that sample.
```
export const MSAL_CONFIG = new InjectionToken<BrowserAuthOptions>('MSAL_CONFIG');

function msalInstanceFactory(authConfig: BrowserAuthOptions): PublicClientApplication {
  return new PublicClientApplication({
    auth: authConfig,
    cache: {
      cacheLocation: BrowserCacheLocation.LocalStorage,
      storeAuthStateInCookie: false
    }
  });
}

export function provideAuthServices(authConfig: BrowserAuthOptions, interceptorConfig: MsalInterceptorConfiguration, guardConfig: MsalGuardConfiguration): EnvironmentProviders {
  return makeEnvironmentProviders([
    { provide: HTTP_INTERCEPTORS, useClass: MsalInterceptor, multi: true },
    { provide: MSAL_INSTANCE, useFactory: () => msalInstanceFactory(authConfig) },
    { provide: MSAL_CONFIG, useValue: authConfig },
    { provide: MSAL_GUARD_CONFIG, useValue: guardConfig },
    { provide: MSAL_INTERCEPTOR_CONFIG, useValue: interceptorConfig },
    MsalService,
    MsalGuard,
    MsalBroadcastService,
    { provide: AuthService, useClass: AuthService }
  ]);
};
```

**Comment by EmLauber** [comment]
For those that experienced yarn or other package manager issues, [v3 alpha-1](https://github.com/AzureAD/microsoft-authentication-library-for-js/releases/tag/msal-angular-v3.0.0-alpha.1) is out and should address those problems.

> no option for selecting v3 of MSAL-Angular.

We are updating our drop downs. For now you can just enter the version as v3 alpha in the version field for the wrapper. 

> make "standalone" sample

I'll pass along this feedback. Our priority is validating v3 and existing sample works with Angular 16, then we can iterate on other samples for Angular 16.

**Comment by grosch-intl** [comment]
> For those that experienced yarn or other package manager issues, [v3 alpha-1](https://github.com/AzureAD/microsoft-authentication-library-for-js/releases/tag/msal-angular-v3.0.0-alpha.1) is out and should address those problems.

Thank you!

**Comment by dansau84** [comment]
what is the solution or partial workaround for this error about angular and msal version incompatibility.

**Comment by grosch-intl** [comment]
@dansau84 The temporary workaround I'm doing is just adding this to my package.json:

```json
  "dependencies": {
    "@azure/msal-angular": "3.0.0-alpha.2",
    "@azure/msal-browser": "3.0.0-alpha.2",
    ...
  },
  "resolutions": {
    "@azure/msal-browser": "3.0.0-alpha.2"
  },
  "overrides": {
    "@azure/msal-browser": "3.0.0-alpha.2"
  }
```

That way both npm and yarn will grab that specific version I want.

**Comment by alexander-kastil** [comment]
Not having an msal package that is ivy compatible is a showstopper for Angular 16 migration as ngcc was removed. Microsoft is aware of this problem for at least 3-4 months. Still you seem to have all the time in the world … I really have to state that this is a very poor performance for an enterprise auth system like Azure AD … verry disappointing 🤬

**Comment by alex737876** [comment]
@EmLauber Could you please give a short feedback on the current state? :)

**Comment by EmLauber** [comment]
Ivy support is available in our latest major release, MSAL.Angular v3. The release is still in alpha as we gather feedback and we are [releasing updates](https://github.com/AzureAD/microsoft-authentication-library-for-js/releases/tag/msal-angular-v3.0.0-alpha.2) with that feedback. You can help us by testing v3 alpha and creating [new issues](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/new/choose) for any bugs you find. I am leaving this issue about Angular 15 support open until v3 is out of pre-release.

**Comment by grosch-intl** [comment]
@EmLauber Any chance of getting faster releases? For an alpha, I was hoping for more like weekly pushes.

**Comment by EricBichara** [comment]
Anyone get v3 working with a standalone app? I'm getting issues as soon as i add the MsalInterceptor using:

`{
        provide: HTTP_INTERCEPTORS,
        useClass: MsalInterceptor,
        multi: true,
      },
      provideHttpClient(withInterceptorsFromDi()),`

**Comment by grosch-intl** [comment]
@EricBichara `provideHttpClient` doesn't work with this package yet.  You need to still add `HttpClientModule` to the `importProvidersFrom` call.  See issue #5988

**Comment by Marcel0024** [comment]
The angular v16 sample still uses the old way `.platformBrowserDynamic()` in main.ts

**Comment by dzhavat** [comment]
Just updated our app to `3.0.0-beta.0` today and everything still works. We're on Angular v15.

**Comment by ShadowNightCoder** [comment]
when will be published the new @azure/msal-angular library update that isn't an alpha that supports angular 15 and 16?

**Comment by SirMrJames** [comment]
I think there needs to be a push to get a fully working production release of this. Not supporting Angular 16 is one thing, but Angular 15 was released 8 months ago.

**Comment by liam-langstaff** [comment]
+1 for [SirMrJames](https://github.com/SirMrJames). V16 Bombs out meaning I'll have to remain on (equally an unsupported) v15 due to this.

**Comment by devfservant** [comment]
Could you please give us some hindsights of when a stable v3 version will be released?

As already mentioned [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/4298#issuecomment-1633929402), using a beta version is a showstopper for some companies.  
It prevents them from upgrading to Angular 15.  
On the other hand, Angular 14's security support ends in 4 months: https://endoflife.date/angular.

FYI, we tested MSAL `3.0.0-beta.1` and it works fine.

**Comment by jerone** [comment]
I raised this issue 8 months ago (tomorrow exactly) for Angular 15, almost a month before it's official release.

For a huge company like Microsoft, for a product that screams security, this is a huge let down. 😢

How many of Microsoft products use this library, I'm scared now.

**Comment by alexander-kastil** [comment]
@devfservant: I also switched from Angular 15 and MSAL 2.x to Angular 16 and MSAL 3.0.0-beta.1 and it worked with no problem. But I agree with @jerone: Do we want to deploy beta versions to production? Shame on you MSAL Team!!!

**Comment by talgershman** [comment]
any news when beta will be over ?

**Comment by alexander-kastil** [comment]
@devfservant If you are hosting on azure you could use easy auth instead of msal … supported on static web app, container apps, app services, … it also allows you to get hold of id- and access tokens … 

https://learn.microsoft.com/en-us/azure/static-web-apps/authentication-authorization

https://learn.microsoft.com/en-us/azure/container-apps/authentication

**Comment by achrysanthakopoulou** [comment]
Version 3.0.0 just released, however I still get the same error when on Angular 16. (Cannot find module '@azure/msal-angular' or its corresponding type declarations.)
Anyone?

**Comment by greut** [comment]
@achrysanthakopoulou `-beta.1` is fine, `3.0.0` however is not, wait a bit.

**Comment by jo-arroyo** [comment]
Hi all, there was an issue with our release process, and we are currently working on a fix. Thank you all for your patience.

**Comment by jerone** [comment]
@jo-arroyo Have all issues, regarding support for Angular 15 & 16, officially been resolved?

**Comment by EmLauber** [comment]
Yes, the releases issues were resolved with the patch v3.0.1. I am closing this issue as support for Angular 15 & 16 is now GA. Please see our [migration guide](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/docs/v2-v3-upgrade-guide.md) for MSAL.Angular v2 to v3.
